src/components/RecipeSubmit.vue

```diff
@@ -0,0 +1,297 @@
+  import VTextField from "vuetify/es5/components/VTextField/VTextField";
+
+  export default {
+    components: {
+      VTextField,
+      VSelect,
+      MdLayout
+    },
+    name: 'recipeSubmit',
+    data () {
+      return {
+        showCarousel: true,
+        localPhotos: [], // Photos stored in the browser
+
+        recipeName: undefined,
+        recipeDescription: undefined,
+        ingredients: [],
+        ingredientOptions: [],
+        prepTimeHr: undefined,
+        prepTimeMin: undefined,
+        cookTimeHr: undefined,
+        cookTimeMin: undefined,
+        servings: undefined,
+        cookware: [],
+        cookwareOptions: [],
+        photos: [],
+        tags: [],
+        recipeSteps: [
+          {}
+        ]
+      };
+    },
+    mounted: function() {
+      // Load cookware dropdown options
+      var cookware = [];
+      db.ref('cookware/').orderByChild('description').on('child_added', function(c) {
+        cookware.push(c.val().description);
+      });
+      this.cookwareOptions = cookware;
+
+      // Load ingredient dropdown options
+      var ingredients = [];
+      db.ref('ingredients/').orderByChild('name').on('child_added', function(i) {
+        ingredients.push(i.val().name);
+      });
+      this.ingredientOptions = ingredients;
+    },
+    methods: {
+      onFileChange(e) {
+        this.showCarousel = false; // Reinitialization hack because dynamic images won't be supported until v1.0
+
+        var files = e.target.files || e.dataTransfer.files;
+        if (!files.length) {
+          return;
+        }
+
+        var storageRef = Firebase.storage().ref(); // TODO: How do I put this in database.js like we did with db?
+
+        var fileList = [];
+        var photoList = [];
+
+        for (var i = 0; i < files.length; i++) {
+          fileList.push({src: window.URL.createObjectURL(files[i])});
+
+          var uploadTask = storageRef.child('images/recipes/' + uuidV4()).put(files[i]);
+          uploadTask.on('state_changed', function(snapshot) {
+            // TODO: Show upload progress?
+          }, function(error) {
+            // TODO: Show snackbar
+          }, function() { // Success
+            // TODO: Show snackbar
+            photoList.push(uploadTask.snapshot.downloadURL); // TODO: Might want to store ref instead of just retrieval URL
+          });
+        }
+        this.localPhotos = fileList;
+        this.photos = photoList;
+        this.$nextTick(() => (this.showCarousel = true));
+      },
+      // TODO: Handle undefineds since Firebase will reject the whole request if they're present.
+      submitRecipe() {
+        db.ref().child('recipes').push().set({
+          submitter: Firebase.auth().currentUser.uid,
+          name: this.recipeName,
+          description: this.recipeDescription,
+          ingredients: this.ingredients.filter(x => x.trim() != ''),
+          prepTime: (parseInt(this.prepTimeHr) * 60 + parseInt(this.prepTimeMin)),
```

Comment:
Explicitly parenthesize arithmetic to make it easier to read. Ex:
prepTime: (parseInt(this.prepTimeHr) * 60) + parseInt(this.prepTimeMin),